src/revwalk.c

```diff
@@ -203,6 +203,11 @@ int git_revwalk_push_range(git_revwalk *walk, const char *range)
 	if ((error = push_commit(walk, git_object_id(revspec.from), 1, false)))
 		goto out;
 
+	if (!revspec.to) {
+		git_error_set(GIT_ERROR_INVALID, "no revspec to found needs to be in format FROM..TO");
+		return GIT_EINVALIDSPEC;
+	}
```

Comment:
I agree that this fix is needed, and @ethomson is probably right in that there ought to be the same check for `revspec.from`. I'd move up this check though to be located directly after the `git_revparse` call: first do all the error checking, and only afterwards should we do anything. As right now, we may end up in an inconsistent state where we've pushed `revspec.from` but then afterwards failed due to `revspec.to` not being set.

Reply:
I got a mild stroke reading that error sentence. Could use some punctuation ;) or something like `invalid revspec`